src/core/lib/layer.js

```diff
@@ -388,17 +381,29 @@ export default class Layer {
       model.geometry.id = `${this.props.id}-geometry`;
       model.setAttributes(attributeManager.getAttributes());
     }
+
+    this.clearChangeFlags();
   }
 
-  // Called by layer manager when existing layer is getting new props
-  updateLayer(updateParams) {
-    // Check for deprecated method
-    if (this.shouldUpdate) {
-      log.deprecated('shouldUpdate', 'shouldUpdateState');
-    }
+  // Called by layer manager
+  // if this layer is new (not matched with an existing layer) oldProps will be empty object
+  updateLayer({oldProps = {}, oldContext = {}}) {
+    this.oldProps = oldProps;
```

Comment:
Is this used anywhere? Shouldn't layer manager be the only source of truth?

Reply:
> Is this used anywhere? Shouldn't layer manager be the only source of truth?

Assuming you are talking about `oldProps`, I believe I understand your point. 

This is my thinking: if we want to enable a delayed call to `updateState` which is part of what this PR paves the way for, we will need to be able to serve `oldProps` to the function at that time, when the layer manager may no longer be involved.

My intention is that oldProps should be set on the layer by the layer manager during layer matching and then it would never change again. Then the layer manager should not need to keep sending it in.  I am wary of doing to much in one PR though.